### PR TITLE
refactor: atomic select #P23-382 

### PR DESCRIPTION
**Pull request by wyola**
## Atomic Select component
task: https://tracker.intive.com/jira/browse/PATRO23-382

**Thank you to @aniasgit @Vectari and @AzulloCyll who created original selects and I could do the reversed engineering based on them 🌷** 

Everything works the same as was, so not a great change for the browser audience 😅 but to sum up:
- 9 files and 221 lines of code less to maintain
- no more styles copied separately in each select
- currency Select now uses the dictionary, so currencies are in all languages (hardcoded in English previously)
- ... hopefully will be easily used with upcoming needs 👀 🤞 if you spot any possible problems, please let me know!

## New feature :sparkles:
I added the possibility of scrolling the content. The scrolling buttons show up when there is not enough space on the screen. I think it might become useful when the custom categories will come.

What would happen previously:

select_duzo opcji

What will happen now:

https://github.com/intive/patronage2023-js/assets/80450315/611314c5-3e25-4966-8f55-7e36ee4aca3e

~~## TODO~~ **DONE!**
~~While writing this, I realized there is one more select in pagination 😅 I will change it soon!~~

## UPDATE
While working on pagination select I noticed that pagination on the user's page is not at the bottom and it looks a bit weird IMO ⬇️ 

image

The last commit includes changes to fix this. I clicked through the page and I don't see that it disrupts anything, but if you could also check it would be great!

### CODE REVIEW

**Comment by aniasgit** [comment]
Resolve conflicts

**Review by Vectari — APPROVED** [review]
Really good job! Everything works great and looks good to me. 🚀

**Review by monika-klokosinska-intive — APPROVED** [review]

**Review by AzulloCyll — APPROVED** [review]
🚀

**Review by bdeluga — APPROVED** [review]
I hope there will be chance to use it ! 🚀

**Review by Noshi96 — APPROVED** [review]
It works much better when the selects are standardized  🚀✨

**Review by aniasgit — APPROVED** [review]
good job 👍

File: packages/ui/Select/Select.styles.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  background-color: ${({ theme }) => theme.select.background};
+  border: solid 2px ${({ theme }) => theme.select.border};
+  border-radius: 8px;
+  font-size: 1em;
+  cursor: pointer;
+  transition: border-color 200ms ease-out;
+  position: relative;
+
+  :focus {
+    transition: border-color 200ms ease-out;
```
[the remark this answers is not in the document]
Comment (wyola) [reply packages/ui/Select/Select.styles.ts]: @Noshi96 thanks for noticing! especially that thanks to this I found that select is not focusing when clicked and fixed it as well

**Pull request merged**